src/main/java/com/sforce/cd/apexUnit/client/connection/ConnectionHandler.java

```diff
@@ -106,21 +115,16 @@ public static ConnectionHandler getConnectionHandlerInstance() {
 			
 	private PartnerConnection createConnection() {
 		if (connection == null) {
-			ConnectorConfig config = new ConnectorConfig();
-			config.setUsername(CommandLineArguments.getUsername());
-			config.setPassword(CommandLineArguments.getPassword());
+			ConnectorConfig config = createConfig();
 			config.setAuthEndpoint(CommandLineArguments.getOrgUrl() + "/services/Soap/u/" + SUPPORTED_VERSION);
-			
-			if (CommandLineArguments.getProxyHost() != null &&  CommandLineArguments.getProxyPort() !=null ){
-				LOG.debug("Setting proxy configuraiton to " + 
-					CommandLineArguments.getProxyHost() + 
-					" on port " + CommandLineArguments.getProxyPort() );
-				config.setProxy( CommandLineArguments.getProxyHost() , CommandLineArguments.getProxyPort());
-			}	
-			
+			config.setSessionRenewer(new SFDCSessionRenewer());
```

Comment:
Khan was unable to look into the test scripts before he left. You may have to add those test cases up on your return. @adarsh-ramakrishna-sfdc